inst/www/shared/shiny.js

```diff
@@ -5006,6 +5006,41 @@ var _typeof = typeof Symbol === "function" && typeof Symbol.iterator === "symbol
     e.preventDefault();
   });
 
+  exports.addCustomMessageHandler("shiny-insert-ui",
+    function(message) {
+      let targets = $(message.selector);
+      if (targets.length === 0) {
+        // render the HTML and deps to a null target, so
+        // the side-effect of rendering the deps, singletons,
+        // and <head> still occur
+        exports.renderHtml(
+          $([]),
+          message.content.html,
+          message.content.deps
+        );
+      } else {
+        targets.each((i, target) => {
+          let container = document.createElement(message.container);
+          target.insertAdjacentElement(message.where, container);
+          exports.renderContent(container, message.content);
+          $(container).trigger("shown");
+          return message.multiple;
+        });
+      }
+    }
+  );
+
+  exports.addCustomMessageHandler("shiny-remove-ui",
+    function(message) {
+      let els = $(message.selector);
+      els.each((i, el) => {
+        $(el).remove();
```

Comment:
We need to do something here to unbind any inputs/outputs before they are removed. I'm... actually not exactly sure what the right way to do this is. Definitely `exports.unbindAll(el)` is part of it, but there's an issue if `el` itself is an input or output--I don't think it'll be unbound in that case. We might need to take each element we're about to remove, and move it into a temporary div, which we then call unbindAll on... pretty gross I guess...